# Objetos de Bioconductor para datos de expresión

## SummarizedExperiment

<iframe width="560" height="315" src="https://www.youtube.com/embed/lqxtgpD-heM" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

[_Original notes in English_](https://docs.google.com/document/d/1umDODmdQldf5w2lNDoFe-unmezHPonpCiKD270VwkrQ/edit?usp=sharing)

* `rnaseqGene` workflow
    - http://bioconductor.org/packages/release/BiocViews.html#___Workflow
    - http://bioconductor.org/packages/release/workflows/html/rnaseqGene.html
    - http://bioconductor.org/packages/release/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html

* `SummarizedExperiment`
    - https://www.nature.com/articles/nmeth.3252 Figure 2
    - http://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html
    - Introducción a `SummarizedExperiment`:     - http://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html
    - Extendiendo `SummarizedExperiment`: http://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/Extensions.html

* `GenomicRanges`
    - http://bioconductor.org/packages/release/bioc/html/GenomicRanges.html
    - https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003118
    - Introducción a `GenomicRanges`: http://bioconductor.org/packages/release/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.html

* `rtracklayer`
    - http://bioconductor.org/packages/release/bioc/html/rtracklayer.html
    - http://bioconductor.org/packages/release/bioc/vignettes/rtracklayer/inst/doc/rtracklayer.pdf
    - Formato de archivos BED https://genome.ucsc.edu/FAQ/FAQformat.html#format1
    - Formato de archivos GTF https://useast.ensembl.org/info/website/upload/gff.html
    
```{r first_rse}
## Lets build our first SummarizedExperiment object
library("SummarizedExperiment")
## ?SummarizedExperiment


## De la
nrows <- 200
ncols <- 6
counts <- matrix(runif(nrows * ncols, 1, 1e4), nrows)
rowRanges <- GRanges(
    rep(c("chr1", "chr2"), c(50, 150)),
    IRanges(floor(runif(200, 1e5, 1e6)), width = 100),
    strand = sample(c("+", "-"), 200, TRUE),
    feature_id = sprintf("ID%03d", 1:200)
)
colData <- DataFrame(
    Treatment = rep(c("ChIP", "Input"), 3),
    row.names = LETTERS[1:6]
)
rse <- SummarizedExperiment(
    assays = SimpleList(counts = counts),
    rowRanges = rowRanges,
    colData = colData
)
rse
dim(rse)
dimnames(rse)
assayNames(rse)
head(assay(rse))

rowRanges(rse)
rowData(rse) # same as 'mcols(rowRanges(rse))'
colData(rse)
```

